tests/utils/mock.py

```diff
@@ -0,0 +1,32 @@
+import contextlib
+import os
+import sys
+from os.path import join, isfile
+from typing import Dict
+
+
+class ProgramMock:
+    def __init__(self, directory: str):
+        self.directory = str(os.path.abspath(directory))
+        os.makedirs(self.directory, exist_ok=True)
+
+    def update_env(self, env: Dict[str, str]):
+        path = ""
+        if "PATH" in env:
+            path = f"{env['PATH']}:"
+        path += self.directory
```

Comment:
Isnt it better to put mocked directory at the beginning of the path? What is qsub is actually installed on machine?